src/sql/workbench/browser/modelComponents/button.component.ts

```diff
@@ -51,43 +59,48 @@ export default class ButtonComponent extends ComponentWithIconBase<azdata.Button
 
 	ngOnInit(): void {
 		this.baseInit();
+
 	}
 
 	ngAfterViewInit(): void {
 		if (this._inputContainer) {
 			this._button = new Button(this._inputContainer.nativeElement);
+		}
 
-			this._register(this._button);
-			this._register(attachButtonStyler(this._button, this.themeService, {
-				buttonBackground: SIDE_BAR_BACKGROUND, buttonHoverBackground: SIDE_BAR_BACKGROUND, buttonForeground: SIDE_BAR_TITLE_FOREGROUND
-			}));
-			this._register(this._button.onDidClick(e => {
-				if (this._fileInputContainer) {
-					const self = this;
-					this._fileInputContainer.nativeElement.onchange = () => {
-						let file = self._fileInputContainer.nativeElement.files[0];
-						let reader = new FileReader();
-						reader.onload = (e) => {
-							let text = (<FileReader>e.target).result;
-							self.fileContent = text.toString();
-							self.fireEvent({
-								eventType: ComponentEventType.onDidClick,
-								args: {
-									filePath: file.path,
-									fileContent: self.fileContent
-								}
-							});
-						};
-						reader.readAsText(file);
-					};
-				} else {
-					this.fireEvent({
-						eventType: ComponentEventType.onDidClick,
-						args: e
-					});
-				}
-			}));
+		if (this._informationalInputContainer) {
+			this._button = new InfoButton(this._informationalInputContainer.nativeElement);
 		}
+
+		this._register(this._button);
+		this._register(attachButtonStyler(this._button, this.themeService, {
+			buttonBackground: SIDE_BAR_BACKGROUND, buttonHoverBackground: SIDE_BAR_BACKGROUND, buttonForeground: SIDE_BAR_TITLE_FOREGROUND
+		}));
+		this._register(this._button.onDidClick(e => {
+			if (this._fileInputContainer) {
+				const self = this;
+				this._fileInputContainer.nativeElement.onchange = () => {
+					let file = self._fileInputContainer.nativeElement.files[0];
+					let reader = new FileReader();
+					reader.onload = (e) => {
+						let text = (<FileReader>e.target).result;
```

Comment:
(e.target as FileReader), the diamond notation is deprecated

Reply:
Fixed